File: leetcodepractice/345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseVowels = function(s) {
    const vowels = 'aeiou';
    const splitInput = s.split('');
    let originalOrder = []

    for (let i = 0; i < s.length; i++) {
        if (vowels.includes(s[i].toLowerCase())) originalOrder.push(s[i]);
    }

    originalOrder = originalOrder.reverse();
    let oI = 0;

    while (oI < originalOrder.length) {
        for (let i = 0; i < s.length; i++) {
            if (vowels.includes(s[i].toLowerCase())) {
                splitInput[i] = originalOrder[oI];
                oI++;
            }
        }
    }

    return splitInput.join('');
};

File: 345-reverse-vowels-of-a-string/reverse-vowels-of-a-string.py
class Solution:
    def reverseVowels(self, s: str) -> str:
        l = 0
        r = len(s) - 1
        vowels = "aeiouAEIOU"
        s_array = list(s)

        while (l < r):
            if s_array[l] in vowels and s_array[r] in vowels:
                s_array[l], s_array[r] = s_array[r], s_array[l]
                l += 1
                r -= 1
            elif s_array[l] in vowels and s_array[r] not in vowels:
                r -= 1
            else:
                l += 1
        
        return "".join(s_array)
                
